metagraph/src/graph/representation/succinct/dbg_succinct.cpp

```diff
@@ -299,18 +289,34 @@ ::call_nodes_with_suffix_matching_longest_prefix(
     );
 
     if (std::get<0>(index_range) == 0 || std::get<1>(index_range) == 0)
-        return;
+        return {};
 
     if (std::get<2>(index_range) == encoded.begin()) {
-        callback(std::get<0>(index_range), 0);
-        return;
+        return {};
     }
 
+    return { std::get<0>(index_range),
+             std::get<1>(index_range),
+             std::get<2>(index_range) - encoded.begin() };
+}
+
+void DBGSuccinct
+::call_nodes_with_suffix_matching_longest_prefix(
+            std::string_view str,
+            std::function<void(node_index, uint64_t /* match length */)> callback,
+            size_t min_match_length,
+            size_t max_num_allowed_matches) const {
+    if (!max_num_allowed_matches || str.size() < min_match_length)
+        return;
```

Comment:
If you call this function just in one place, remove it.

Reply:
it's used in a few places (seeding, CanonicalDBG, align.cpp)